server/services/participants.js

```diff
@@ -107,36 +107,43 @@ const getHiredParticipantsBySite = async (siteID) => {
   return participants;
 };
 
+/**
+ * Complexity in this function relates to withdrawal reason being stored in a record separate
+ * from the participant's site.
+ *
+ * 1. Find previously hired participants for a site
+ * 2. Get list of archived candidates with IDs that match previously hired participants
+ *
+ * @param {string} siteID
+ * @returns list of withdrawn participants+status related to a given siteID
+ */
 const getWithdrawnParticipantsBySite = async (siteID) => {
-  const participants = await dbClient.db[collections.PARTICIPANTS_STATUS]
-    .join({
-      participantJoin: {
-        type: 'LEFT OUTER',
-        relation: collections.PARTICIPANTS,
-        decomposeTo: 'object',
-        on: { id: 'participant_id' },
-      },
-    })
-    .find({
-      current: true,
-      status: 'archived',
-    });
+  const participantsStatusJoin = dbClient.db[collections.PARTICIPANTS_STATUS].join({
```

Comment:
Can you try this 
```js
const join = db[collections.PARTICIPANTS_STATUS].join({
        selfArchived: {
          type: 'JOIN',
          relation: collections.PARTICIPANTS_STATUS,
          on: {
            participant_id: 'participant_id',
            status: 'archived',
            current: true,
          }
        },
        participantJoin: {
          type: 'LEFT OUTER',
          on: {
            id: 'participant_id'
          }
        }
      }).find({
        status: 'hired',
        current: false,
        'data.site': siteID
      })
```

Not sure this will work or not possibly I try raw query to check.

Reply:
Awesome suggestion to try a self join, having troubles reproducing the functionality in MassiveJS so I'll table it for now